components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.edit/src-gen/org/wso2/integrationstudio/gmf/esb/presentation/condition/manager/EnableConditionManager.java

```diff
@@ -83,5 +87,25 @@ private void setComponentVisibility(EnableCondition enableCondition, boolean vis
             composite.getParent().getParent().layout();
         }
     }
+    
+    private void addToDependentControls(EnableCondition enableCondition, boolean visibility) {
+    	String targetComponentName = enableCondition.getTargetComponentName();
+        if (compositeList.containsKey(targetComponentName)) {
+            Composite composite = compositeList.get(targetComponentName);
+            if (!visibility) {
+            	if (!dependentInvisibleComposites.containsKey(composite)) {
+            		dependentInvisibleComposites.put(composite, visibility);
+            	}
```

Comment:
Can we use `putIfAbsent`?

Reply:
Yes. Fixed in https://github.com/wso2/integration-studio/pull/1167/commits/aff499587d662c8ba7e54cb69cf443367a88d7e3